modal/image.py

```diff
@@ -266,6 +278,47 @@ def ignore_with_include(source: Path) -> bool:
     return _Mount._add_local_dir(Path("./"), PurePosixPath("/"), ignore=ignore_with_include)
 
 
+def _create_context_mount_function(
+    ignore: Union[Sequence[str], Callable[[Path], bool]],
+    dockerfile_cmds: list[str] = [],
+    dockerfile_path: Optional[Path] = None,
+    context_mount: Optional[_Mount] = None,
+):
+    if dockerfile_path and dockerfile_cmds:
+        raise InvalidError("Cannot provide both dockerfile and docker commands")
+
+    if context_mount:
+        if ignore is not AUTO_DOCKERIGNORE:
+            raise InvalidError("Cannot set both `context_mount` and `ignore`")
+
+        def identity_context_mount_fn() -> Optional[_Mount]:
+            return context_mount
+
+        return identity_context_mount_fn
+    elif ignore is AUTO_DOCKERIGNORE:
+
+        def auto_created_context_mount_fn() -> Optional[_Mount]:
+            context_dir = Path.cwd()
+            dockerignore_file = find_dockerignore_file(context_dir, dockerfile_path)
+            ignore_fn = (
+                FilePatternMatcher(*dockerignore_file.read_text("utf8").splitlines())
+                if dockerignore_file
+                else _ignore_fn(())
+            )
+
+            cmds = dockerfile_path.read_text("utf8").splitlines() if dockerfile_path else dockerfile_cmds
+            return _create_context_mount(cmds, ignore_fn=ignore_fn, context_dir=context_dir)
+
+        return auto_created_context_mount_fn
+
+    def auto_created_context_mount_fn() -> Optional[_Mount]:
+        # use COPY commands and ignore patterns to construct implicit context mount
+        cmds = dockerfile_path.read_text("utf8").splitlines() if dockerfile_path else dockerfile_cmds
```

Comment:
This is a bit more general than this PR and could be adressed in a followup - but can't `COPY` commands also be split over multiple lines using `\` ?

Reply:
Yeah we actually added multiline support for `COPY` commands specifically: https://github.com/modal-labs/modal-client/blob/main/modal/_utils/docker_utils.py#L25C1-L33C34 but for other commands (like `RUN`) we don't support multiline commands, we should add that!

Reply:
Maybe we can refactor this cmd reading to combine multilines directly instead of relying on the underlying `extract_copy_command_patterns` function to do it for `COPY` commands